docs/Scaling.md

```diff
@@ -0,0 +1,58 @@
+# When Scaling is needed?
+
+While our clusters keep growing in size, there will be more data gathered by Prometheus and at some point we can reach the maximum memory of the node; most of the time related to the cardinality of the time series.
+In this situation you can observe that your Prometheus servers start dying (ideally this should be seen before) because they need more memory. This is an irrefutable sign that you need to face scaling.
+
+In order to analyze the solution in detail, we provide a dashboard with different charts that will help us know when we have to scale our Prometheus solution.
+
+There are two different scaling approaches for `newrelic-prometheus`: 
+
+- Vertical Scaling
+
+- Horizontal Scaling
+
+## Vertical Scaling
+
+This kind of scaling comes without any complexity. It's as simple as updating the memory and/or the cpu for the corresponding machine where cluster the node is living.
+
+However, this approach may not be scalable for huge clusters, or simply we just don't want to have a single pod that consumes so many GBs of memory in our node. If so, you may need to take at <b>Horizontal Scaling<b/>.
+
+## Horizontal Scaling (Sharding)
+
+Horizontal Scaling is supported by setting up a configuration parameter which allows running several prometheus servers in agent mode to gather your data. This is known as Sharding.
+
+If you define `sharding.total_shards_count` value, the deployed StatefulSet will include as many replicas as you defined there. When it is used, the _configurator_ component will automatically include some additional relabel rules so each target will only be scraped by one prometheus server. Those
+rules rely on the target's address [hash-mod](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config).
```

Comment:
i would maybe explain this with a bit more detail, like... each instance computes the this module of a hash and if the result matches the current instance the target is scraped. The selection is mutually exclusive meaning that a target will be only scraped by one shard as long as the shard count and the address of the target doesn't change.

Reply:
nit: perhaps a simple diagram could make the solution really easy to understand too

Reply:
Done in e946479fa0f27027615dd98d00fee9c9f1e6705a